packages/core/src/components/fields/BooleanField.tsx

```diff
@@ -36,14 +36,18 @@ function BooleanField<T = any, S extends StrictRJSFSchema = RJSFSchema, F extend
     rawErrors,
   } = props;
   const { title } = schema;
-  const { widgets, formContext, translateString } = registry;
-  const { widget = 'checkbox', ...options } = getUiOptions<T, S, F>(uiSchema);
+  const { widgets, formContext, translateString, globalUiOptions } = registry;
+  const {
+    widget = 'checkbox',
+    title: uiTitle,
+    label: displayLabel = true,
```

Comment:
```suggestion
     // Unlike the other fields, we don't want to use `getDisplayLabel()` since it always returns false for the boolean type
    label: displayLabel = true,
```